protected/humhub/modules/comment/controllers/CommentController.php

```diff
@@ -147,15 +147,7 @@ public function actionEdit()
         $comment = Comment::findOne(['id' => Yii::$app->request->get('id')]);
 
         // Find all file attached to this comment
-        $files = $comment->fileManager->find()->asArray()->all();
-        $fileList = [];
-        $i = 0;
-
-        // Fill up files array
-        foreach ($files as $file) {
-            $fileList[$i] = $file['guid'];
-            $i++;
-        }
+        $fileList = $comment->fileManager->findAssignedToRecord();
```

Comment:
Can you explain, why we need this here? I assume only because of the validation, since we do not have to reassign those files once they are assigned? In case you only need it for validation wouldn't a `$comment->fileManager->find()->all()` be sufficient?

Reply:
I will fix it you are totally right. Thank's for your comment. We do not need to check for a new or old file version. We simply need to know is there any files attached or not.